packages/itwinui-react/src/core/Buttons/IconButton.tsx

```diff
@@ -4,14 +4,16 @@
  *--------------------------------------------------------------------------------------------*/
 import cx from 'classnames';
 import * as React from 'react';
-import { Box, ButtonBase } from '../../utils/index.js';
+import { Box, ButtonBase, createWarningLogger } from '../../utils/index.js';
 import { Tooltip } from '../Tooltip/Tooltip.js';
 import type { ButtonProps } from './Button.js';
 import type { PolymorphicForwardRefComponent } from '../../utils/index.js';
 import { VisuallyHidden } from '../VisuallyHidden/VisuallyHidden.js';
 import { ButtonGroupContext } from '../ButtonGroup/ButtonGroup.js';
 import { PopoverOpenContext } from '../Popover/Popover.js';
 
+const logWarning = createWarningLogger();
```

Comment:
There is a slight issue with our `createWarningLogger` utility. It only runs once. Not once per instance of component but once per the file where it is called.

This issue is not specific to this PR, but in the future we should fix it to run once per instance of a component, so that two different `<IconButton>`s will trigger two different warnings.